lib/index.js

```diff
@@ -164,12 +142,15 @@ module.exports = {
       }
     }
 
-    this._jqueryIntegrationEnabled =
-      optionalFeaturesMissing || optionalFeatures.isFeatureEnabled('jquery-integration');
-
-    if (this._jqueryIntegrationEnabled) {
-      this.ui.writeWarnLine(
-        'Setting the `jquery-integration` optional feature flag to `true`, or not providing a setting at all, has been deprecated. You must add the `@ember/optional-features` addon and set this feature to `false`. This warning will become an error in Ember 4.0.0.\n\nFor more information, see the deprecation guide: https://deprecations.emberjs.com/v3.x/#toc_optional-feature-jquery-integration'
+    if (
+      !optionalFeaturesMissing &&
+      optionalFeatures.isFeatureEnabled('jquery-integration') &&
+      typeof optionalFeatures.isFeatureExplicitlySet === 'function' &&
+      optionalFeatures.isFeatureExplicitlySet('jquery-integration')
+    ) {
+      const SilentError = require('silent-error');
+      throw new SilentError(
+        'Setting the `jquery-integration` optional feature flag to `true` was deprecated in Ember 3.x and removed in Ember 4.0.0. You must add the `@ember/optional-features` addon and set this feature to `false`.\n\nFor more information, see the deprecation guide: https://deprecations.emberjs.com/v3.x/#toc_optional-feature-jquery-integration'
```

Comment:
I think this prose needs work now that the logic has changed. We don't want folks installing the optional features addon at all, we want them to remove the configuration from their `optional-features.json` file...